fedimint-bin-tests/src/main.rs

```diff
@@ -521,6 +523,401 @@ async fn latency_tests(dev_fed: DevFed) -> Result<()> {
+            ..Default::default()
+        })
+        .await?
+        .into_inner()
+        .state();
+    anyhow::ensure!(invoice_status == tonic_lnd::lnrpc::invoice::InvoiceState::Settled);
+
+    // Assert balances changed by 100000 msat (amount sent) + 1000 msat (fee)
+    let final_client_balance = cmd!(fed, "info").out_json().await?["total_amount"]
+        .as_u64()
+        .unwrap();
+    let final_gateway_balance = cmd!(gw_cln, "balance", "--federation-id={fed_id}")
+        .out_json()
+        .await?["balance_msat"]
+        .as_u64()
+        .unwrap();
+    anyhow::ensure!(
+        initial_client_balance - final_client_balance == 101_000,
+        "Client balance changed by {}, expected 101000",
+        initial_client_balance - final_client_balance
+    );
+    anyhow::ensure!(
+        final_gateway_balance - initial_gateway_balance == 101_000,
+        "Gateway balance changed by {}, expected 101000",
+        final_gateway_balance - initial_gateway_balance
+    );
+
+    // INCOMING: fedimint-cli receives from LND via CLN gateway
+    let invoice = cmd!(
+        fed,
+        "ln-invoice",
+        "--amount=100000msat",
+        "--description='incoming-over-lnd-gw'"
+    )
+    .out_json()
+    .await?["invoice"]
+        .as_str()
+        .context("invoice must be string")?
+        .to_owned();
+    let payment = lnd
+        .client_lock()
+        .await?
+        .send_payment_sync(tonic_lnd::lnrpc::SendRequest {
+            payment_request: invoice.clone(),
+            ..Default::default()
+        })
+        .await?
+        .into_inner();
+    let payment_status = lnd
+        .client_lock()
+        .await?
+        .list_payments(tonic_lnd::lnrpc::ListPaymentsRequest {
+            include_incomplete: true,
+            ..Default::default()
+        })
+        .await?
+        .into_inner()
+        .payments
+        .into_iter()
+        .find(|p| p.payment_hash == payment.payment_hash.to_hex())
+        .context("payment not in list")?
+        .status();
+    anyhow::ensure!(payment_status == tonic_lnd::lnrpc::payment::PaymentStatus::Succeeded);
+
+    // LND gateway tests
+    fed.use_gateway(&gw_lnd).await?;
+
+    // OUTGOING: fedimint-cli pays CLN via LND gateaway
+    let initial_client_balance = cmd!(fed, "info").out_json().await?["total_amount"]
+        .as_u64()
+        .unwrap();
+    let initial_gateway_balance = cmd!(gw_cln, "balance", "--federation-id={fed_id}")
+        .out_json()
+        .await?["balance_msat"]
+        .as_u64()
+        .unwrap();
+    let invoice = cln
+        .request(cln_rpc::model::InvoiceRequest {
+            amount_msat: AmountOrAny::Amount(ClnRpcAmount::from_msat(100_000)),
+            description: "lnd-gw-to-cln".to_string(),
+            label: "test".to_string(),
+            expiry: Some(1),
+            fallbacks: None,
+            preimage: None,
+            exposeprivatechannels: None,
+            cltv: None,
+            deschashonly: None,
```

Comment:
In any of these places where there a bunch of `None` fields in a `cln_rpc::model::*` struct, you can replace them all with a single `Default::default(),` line. This should make the test a little less verbose.

Reply:
`cln_rpc::model::*` don't implement Default :/

Reply:
Ohhh ... that's annoying

Reply:
Made an issue about this upstream https://github.com/ElementsProject/lightning/issues/6174

Reply:
I get the following error when I try to do this:
> the trait `std::default::Default` is not implemented for `*`
`*` being the `cln_rpc::model` struct I've used